src/project-tests/bar-chart-tests.js

```diff
@@ -1,196 +1,282 @@
+            firstRatio.toFixed(3),
+            ratio.toFixed(3),
+            'The heights of the bars should correspond to the data values '
+          );
         }
-      );
-    });
+      });
 
-    reqNum++;
-    it(`${reqNum}. Each bar's height should accurately represent the data's
-    corresponding GDP`,
-    function() {
-      const bars = document.querySelectorAll('rect.bar');
-      // get the ratio of the first data point to the height of the first bar
-      const firstRatio =
-      bars[0].getAttribute('data-gdp') / bars[0].getAttribute('height');
-      /* iterate through each bar and make sure that its height is consistent
-      with its corresponding data point using the first ratio */
-      /* this test completely validates the bars, but isn\'t very useful to the
-      user, so data-date and data-gdp tests were added for clarity */
-      for (var i = 0; i < bars.length; i++) {
-        var dataValue = bars[i].getAttribute('data-gdp');
-        var barHeight = bars[i].getAttribute('height');
-        var ratio = dataValue / barHeight;
-        assert.equal(
-          firstRatio.toFixed(3),
-          ratio.toFixed(3),
-          'The heights of the bars should correspond to the data values '
+      reqNum++;
+      it(`${reqNum}. The data-date attribute and its corresponding bar should 
+      align with the corresponding value on the x-axis.`,
+      function() {
+        const barsCollection = document.getElementsByClassName('bar');
+        const coordAttr = 'x';
+
+        assert.isAbove(
+          barsCollection.length,
+          0,
+          'there are no elements with the class of "bar" '
+        );
+
+        assert.isTrue(
+          isAxisAlignedWithDataPoints(
+            document.querySelector('#x-axis'),
+            coordAttr,
+            barsCollection,
+            getShapeValueYearBar,
+            getTickValueYear,
+            getShapePositionRectBar,
+            getTickPosition
+          ),
+          'x values don\'t line up with x locations '
         );
-      }
+
+      });
+
+      reqNum++;
+      it(`${reqNum}. The data-gdp attribute and its corresponding bar should
+      align with the corresponding value on the y-axis.`,
+      function() {
+        const barsCollection = document.getElementsByClassName('bar');
+        const coordAttr = 'y';
+
+        assert.isAbove(
+          barsCollection.length,
+          0,
+          'there are no elements with the class of "dot" '
+        );
+
+        assert.isTrue(
+          isAxisAlignedWithDataPoints(
+            document.querySelector('#y-axis'),
+            coordAttr,
+            barsCollection,
+            getShapeValueDecimal,
+            getTickValueThousands,
+            getShapePositionRectBar,
+            getTickPosition
+          ),
+          'y values don\'t line up with y locations '
+        );
+
+      });
+
     });
+    testToolTip(document.querySelectorAll('.bar'), 'data-date', 'data-date');
```

Comment:
Could you please add an inline comment above this so it's clear that it is another set of tests? You can see what I did with the Heat Map tests for an example.